### PR TITLE
Use more idiomatic Kotlin in EventSources and elsewhere

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/CertificatePinner.kt [hunk 1]
```diff
@@ -264,21 +242,6 @@ data class CertificatePinner internal constructor(
...
     }
 
-    override fun equals(other: Any?): Boolean {
-      return other is Pin
-          && pattern == other.pattern
-          && hashAlgorithm == other.hashAlgorithm
-          && hash == other.hash
-    }
-
-    override fun hashCode(): Int {
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/CertificatePinner.kt]: Did you mean to make this a data class? Or is it just not compared or used as a key?

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/CertificatePinner.kt]: I meant to make it a data class.

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/CertificatePinner.kt]: Good catch!

File: okhttp/src/main/java/okhttp3/Handshake.kt [hunk 2]
```diff
@@ -72,33 +72,26 @@ data class Handshake private constructor(
...
+      }
+    }
 
   companion object {
     @Throws(IOException::class)
     @JvmStatic
     fun get(session: SSLSession): Handshake {
-      val cipherSuiteString = session.cipherSuite ?: throw IllegalStateException("cipherSuite == null")
+      val cipherSuiteString =
+          session.cipherSuite ?: throw IllegalStateException("cipherSuite == null")
```
Comment (Egorand) [inline okhttp/src/main/java/okhttp3/Handshake.kt]: nit: `checkNotNull(session.cipherSuite) { "cipherSuite == null" }`

File: okhttp/src/main/java/okhttp3/Handshake.kt [hunk 3]
```diff
@@ -72,33 +72,26 @@ data class Handshake private constructor(
...
-          if (it is X509Certificate) {
-            it.subjectDN.toString()
-          } else {
-            it.type
-          }
-        }
-  }
+  private val Certificate.name: String
+    get() {
+      return when (this) {
```
Comment (Egorand) [inline okhttp/src/main/java/okhttp3/Handshake.kt]: nit: 
```kotlin
get() = when (this) {
  is X509Certificate -> subjectDN.toString()
  else -> type
}
```

File: okhttp/src/main/java/okhttp3/Handshake.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/Handshake.kt]: Fixed.

File: okhttp/src/main/java/okhttp3/Handshake.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/Handshake.kt]: Fixed.

File: okhttp/src/main/java/okhttp3/Handshake.kt (hunk 2 above)
Comment (yschimke) [reply okhttp/src/main/java/okhttp3/Handshake.kt]: That checkNotNull is nice. TIL

**Pull request merged**